models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -205,6 +205,11 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
                     // the local tmp-file to the real storage a bit further down doesn't work, as it has a
                     // check for race-conditions & locking, so it needs to check for the existence of the thumbnail
                     $storage->delete($storagePath);
+
+                    // refresh the thumbnail cache, if the asset modification date is modified
+                    // this is necessary because the thumbnail cache is not cleared automatically
+                    // when the original asset is modified
+                    self::addOrUpdateThumbnailCache($asset, $config, $filename);
```

Comment:
When an asset is updated, the thumbnail cache get's cleared as well, so this shouldn't be necessary imho. 
See https://github.com/pimcore/pimcore/blob/d982fa8d96dc775e49a04017cfa657882cedf148/models/Asset.php#L1606

But we could still do `$asset->getDao()->deleteFromThumbnailCache($config-getName());` here, as it doesn't hurt.

Reply:
done